src/textarea/characterLimitChecker.tsx

```diff
@@ -0,0 +1,45 @@
+import * as React from 'react';
+
+interface Props {
+    radius: number;
+    stroke: string;
+    strokeDasharray: string;
```

Comment:
Can you replace that with a `percantageFilled` number based property? Then we can avoid making the client component creating a style string.